modules/core/src/ocl.cpp

```diff
@@ -1240,6 +1240,8 @@ struct Device::Impl
             vendorID_ = VENDOR_INTEL;
         else if (vendorName_ == "NVIDIA Corporation")
             vendorID_ = VENDOR_NVIDIA;
+        else if (vendorName_ == "RM")
+            vendorID_ = VENDOR_ARM;
```

Comment:
Pattern is very generic.

Anyway, these changes are not required for test fixes. Please revert them.